src/shell/xdg/frame/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! The frame to use with XDG shell window.
+
+use crate::reexports::client::protocol::wl_surface::WlSurface;
+use crate::reexports::protocols::xdg::shell::client::xdg_toplevel::{
+    ResizeEdge, State as WindowState,
+};
+
+pub mod fallback_frame;
+
+/// The interface for the client side decorations.
+pub trait DecorationsFrame: Sized {
+    /// Configurations for this frame.
+    type Config;
```

Comment:
I'd argue the config doesn't really need to be part of the frame trait. The config is going to be dependent on the frame type anyways.

Reply:
I think the goal is to transparently pass it, we might remove it, sure?